lib/_http_outgoing.js

```diff
@@ -574,11 +573,16 @@ OutgoingMessage.prototype._finish = function() {
 OutgoingMessage.prototype._flush = function() {
   if (this.socket && this.socket.writable) {
     var ret;
-    while (this.output.length) {
-      var data = this.output.shift();
-      var encoding = this.outputEncodings.shift();
-      var cb = this.outputCallbacks.shift();
-      ret = this.socket.write(data, encoding, cb);
+    // There might be remaining data in this.output; write it out
+    var outputLength = this.output.length;
+    if (outputLength) {
+      for (var i = 0; i < outputLength; i++) {
+        var cb = this.outputCallbacks[i];
+        ret = this.socket.write(this.output[i], this.outputEncodings[i], cb);
```

Comment:
same comment here, not immediately apparent why `cb` exists if you are only using it once